src/ldp/http/SimpleResponseWriter.ts

```diff
@@ -23,9 +23,8 @@ export class SimpleResponseWriter extends ResponseWriter {
     if (input.description) {
       input.response.setHeader('location', input.description.identifier.path);
       if (input.description.body) {
-        if (input.description.body.metadata.contentType) {
-          input.response.setHeader('content-type', input.description.body.metadata.contentType);
-        }
+        const contentType = input.description.body.metadata.contentType || 'text/plain';
```

Comment:
The order is `ResourceDescription` -> `Representation` -> `RepresentationMetadata` -> `contentType`. So one option would be to decouple the metadata from the representation (which is something I started thinking about since it might be interesting to request metadata without the representation at some point), or to have something else define certain headers in advance.